app/lib/core/models/contingency_plan_info.dart

```diff
@@ -194,7 +194,7 @@ class ContingencyPlanEvent {
   }
 
   ContingencyPlanEvent.fromJSON(Map<String, dynamic> json)
-      : eventDateAndTime = DateTime.parse(json['eventDateAndTime']),
+      : eventDateAndTime = json['eventDateAndTime'].toDate(),
```

Comment:
@holtzmak to test this on iPhone. We'll probably need a switch case as per https://stackoverflow.com/a/55398555